### PR TITLE
Empty group put request 500s

### LINKED ISSUE #843: Empty group / other-noun PUTs respond 200
If you send off `{}` to `PUT /groups`, for example, it will respond 200.

I would argue that this only makes sense for jobs, where the engine is explicitly heartbeating the noun.

### PR DESCRIPTION
**Pull request by hkethi002**
Fixes #843 
### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

**Comment by nagem** [comment]
An empty payload is an improper request, so the more accurate response code would be `400: Bad Request`. I would add a check to group/project/session/acquisition/collection/user PUT endpoints that checks to make sure the payload is not empty as soon as possible in the handler logic.

**Comment by nagem** [comment]
It should really be on all PUT endpoints, feel free to add if you'd like, otherwise that list above is a good start.

File: api/handlers/collectionshandler.py [hunk 1]
```diff
@@ -57,6 +57,8 @@ def put(self, **kwargs):
         mongo_validator, payload_validator = self._get_validators()
 
         payload = self.request.json_body or {}
+        if not payload:
+            self.abort(400, 'Bad Request')
```
Comment (nagem) [inline api/handlers/collectionshandler.py]: "Bad Request" is the title of the 400 error, you can put something more descriptive here. "PUT request body cannot be empty" or something.

**Comment by nagem** [comment]
Looks good after response message is changed to be a bit more descriptive!

**Comment by nagem** [comment]
LGTM!

**Comment by kofalt** [comment]
Thanks Harsha.

**Pull request merged**